crates/core_arch/src/macros.rs

```diff
@@ -44,6 +44,13 @@ macro_rules! static_assert_imm8 {
     };
 }
 
+#[allow(unused)]
+macro_rules! static_assert_rounding {
+    ($imm:ident) => {
+        let _ = $imm == 4 || $imm == 8 || $imm == 9 || $imm == 10 || $imm == 11;
```

Comment:
This doesn't actually assert anything. You need to either call `static_assert` within this macro or create a custom `ValidateConst` struct like the other macros.

Also this is x86-specific, please move it to `x86/macros.rs`.